drivers/soundwire/cadence_master.c

```diff
@@ -1753,6 +1750,9 @@ int sdw_cdns_clock_restart(struct sdw_cdns *cdns, bool bus_reset)
 			dev_err(cdns->dev, "bus failed to exit clock stop %d\n", ret);
 	}
 
+	/* unmask Slave interrupts that were masked when stopping the clock */
+	cdns_enable_slave_interrupts(cdns, true);
```

Comment:
Are we sure about this? At what step in the process is the system actually able to handle an IRQ? I guess my slight concern here is it looks like there have been a lot of changes trying to close windows where an IRQ can occur whilst masked and this appears to open such a window.

Reply:
this patch moves the interrupt enabling from external devices at a later time.

This is also part of the programming guidance, I think it makes sense but I am not sure if it changes anything really.